src/utilities/embeddable-utils.ts

```diff
@@ -0,0 +1,89 @@
+import { v4 as uuidv4 } from "uuid";
+import { IAuthoringMetadata, IRuntimeMetadata } from "@concord-consortium/lara-interactive-api";
+import { IExportableAnswerMetadata, IManagedInteractive, IReportState, IExportableMultipleChoiceAnswerMetadata, IExportableOpenResponseAnswerMetadata, IExportableInteractiveAnswerMetadata } from "../types";
+
+export const getAnswerWithMetadata = (
+    interactiveState: IRuntimeMetadata,
+    embeddable: IManagedInteractive,
+    oldAnswerMeta?: IExportableAnswerMetadata): (IExportableAnswerMetadata | void) => {
+
+  if (!embeddable.authored_state) return;
+
+  const authoredState = JSON.parse(embeddable.authored_state) as IAuthoringMetadata;
+  const reportState: IReportState = {
+    mode: "report",
+    authoredState: embeddable.authored_state,
+    interactiveState: JSON.stringify(interactiveState)
+  };
+  if ((authoredState as any).version) {
+    reportState.version = (authoredState as any).version;
+  }
+
+  const reportStateJSON = JSON.stringify(reportState);
+
+  const exportableAnswerBase = {
+    remote_endpoint: "",
+    question_id: refIdToAnswersQuestionId(embeddable.ref_id),
+    question_type: authoredState.questionType,
+    id: oldAnswerMeta ? oldAnswerMeta.id : uuidv4(),
+    type: interactiveState.answerType,
+    answer_text: interactiveState.answerText,
+    submitted: typeof interactiveState.submitted === "boolean" ? interactiveState.submitted : null,
+    report_state: reportStateJSON
+  };
+
+  // from https://github.com/concord-consortium/lara/blob/c40304a14ef495acdf4f9fd09ea892c7cc98247b/app/models/interactive_run_state.rb#L139
+  let exportableAnswer;
+  if (interactiveState.answerType === "multiple_choice_answer") {
+    exportableAnswer = {
+      ...exportableAnswerBase,
+      answer: {
+        choice_ids: interactiveState.selectedChoiceIds
+      }
+    } as IExportableMultipleChoiceAnswerMetadata;
+  } else if (interactiveState.answerType === "open_response_answer") {
+    exportableAnswer = {
+      ...exportableAnswerBase,
+      answer: interactiveState.answerText
+    } as IExportableOpenResponseAnswerMetadata;
+  } else {
+    exportableAnswer = {
+      ...exportableAnswerBase,
+      answer: reportStateJSON
```

Comment:
It would be good to add a comment about how it isn't handling the has_report_url property of interactives. It is correct that this code shouldn't handle it because interactives that use that (mainly CODAP and Sage) would not work in the activity player anyway without additional work. And we are planning to replace their use of the has_report_url with standard interactiveState handling.

Reply:
@sfentress it looks like you missed this request.

Reply:
I see the comment `// note we don't current support has_report_url`. Did you want something further?

Reply:
Sorry I missed that before. I think it was just out of range from 3 line context provided in the github conversation view.